lib/plenario/changesets/changeset_utils.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Plenario.Changesets.Utils do
+  import Ecto.Changeset
+
+  alias Plenario.Actions.MetaActions
+
+  @name_length 16
+
+  def validate_meta_state(%Ecto.Changeset{valid?: true} = changeset) do
+    meta = MetaActions.get(get_field(changeset, :meta_id))
+    case meta.state do
+      "new" -> changeset
+      _ -> add_error(changeset, :base, "Data set is locked")
+    end
+  end
+  def validate_meta_state(changeset), do: changeset
```

Comment:
Ah okay, so it's flat across the board. Can't change meta or anything related to it if it's not "new".

Reply:
Correct.